core/src/test/java/org/elasticsearch/index/mapper/multifield/MultiFieldTests.java

```diff
@@ -536,4 +526,29 @@ public void testNestedFieldNotAllowed() throws Exception {
             assertTrue(e.getMessage().contains("cannot be used in multi field"));
         }
     }
+
+    public void testMultiFieldWithDot() throws IOException {
+        XContentBuilder mapping = jsonBuilder();
+        mapping.startObject()
+                .startObject("my_type")
+                .startObject("properties")
+                .startObject("city")
+                .field("type", "string")
+                .startObject("fields")
+                .startObject("raw.foo")
+                .field("type", "string")
+                .field("index", "not_analyzed")
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject();
+
+        MapperService mapperService = createIndex("test").mapperService();
+        try {
+            mapperService.documentMapperParser().parse(mapping.string());
+            fail("this should throw an exception because one field contains a dot");
+        } catch (MapperParsingException e) {
+        }
```

Comment:
Add an assert that the exception was what we expect (check part of the message)?